webapi/src/main/scala/org/knora/webapi/routing/v1/CkanRouteV1.scala

```diff
@@ -38,21 +39,23 @@ object CkanRouteV1 extends Authenticator {
     def knoraApiPath(_system: ActorSystem, settings: SettingsImpl, log: LoggingAdapter): Route = {
 
         implicit val system: ActorSystem = _system
-        implicit val executionContext = system.dispatcher
-        implicit val timeout = Timeout(30.seconds)
+        implicit val executionContext: ExecutionContextExecutor = system.dispatcher
+        implicit val timeout: Timeout = Timeout(30.seconds)
         val responderManager = system.actorSelection("/user/responderManager")
 
         path("v1" / "ckan") {
             get {
                 requestContext =>
-                    val userProfile = getUserADM(requestContext)
-                    val params = requestContext.request.uri.query().toMap
-                    val project: Option[Seq[String]] = params.get("project").map(_.split(","))
-                    val limit: Option[Int] = params.get("limit").map(_.toInt)
-                    val info: Boolean = params.getOrElse("info", false) == true
-                    val requestMessage = CkanRequestV1(project, limit, info, userProfile)
 
-                    RouteUtilV1.runJsonRoute(
+                    val requestMessage = for {
+                        userProfile <- getUserADM(requestContext)
+                        params = requestContext.request.uri.query().toMap
+                        project: Option[Seq[String]] = params.get("project").map(_.split(","))
+                        limit: Option[Int] = params.get("limit").map(_.toInt)
```

Comment:
I think it's great to make all this code non-blocking. A question: in some cases (like here) you've put all the route processing in a `Future` using a `for` comprehension. But in other cases (like in `GroupsRouteADM`), the request parameters are processed outside the `Future`, and the `Future` is created only at the end of the route. I know that currently, if a route throws an exception, the exception is handled by an Akka exception handler. Does that change now that the code is in a `Future`? For example, on this line, what happens if `toInt` throws an exception? Is the exception handled differently from how it was handled before?

Reply:
good question. just tryied it out:

this call:
```
http://localhost:3333/v1/ckan?limit=one
```

produces this response:

```
{
    "status": 4,
    "error": "java.lang.NumberFormatException: For input string: \"one\""
}
```

So it down's matter if it is thrown here `userProfile <- getUserADM(requestContext)` or here `limit: Option[Int] = params.get("limit").map(_.toInt)`.

Reply:
It is not refactored the same everywhere. After the first few routes, I got tired of doing it and looked for the quickest possible way :-)